### PR TITLE
lib-classifier: Add disableImageToolbar view to SubjectViewerStore, refactor toolbar buttons

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-classifier

## Linked Issue and/or Talk Post
- towards #3093, iteration on #4539

## Describe your changes
- adds `disableImageToolbar` view to SubjectViewerStore
- refactors image related toolbar buttons to be disabled per `disableImageToolbar`
- refactors toolbar buttons styling if disabled
- adds SubjectViewerStore tests for `disableImageToolbar`
- refactor related stories

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
  - local: https://local.zooniverse.org:8080/?project=1693&workflow=3430
  - fe-project-branch: TBD
- What user actions should my reviewer step through to review this PR?
  - load subject viewer with image, note image related buttons work
  - load subject viewer with text or video, note image related buttons are disabled
- Which storybook stories should be reviewed?
  - all ImageToolbar stories
  - [new ImageToolbar story](http://localhost:6006/?path=/story/image-toolbar-image-toolbar--with-single-text-viewer) with text viewer and disabled image related buttons
- Are there plans for follow up PR’s to further fix this bug or develop this feature? maybe, depending on user feedback

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Bug Fix
- [ ] The PR creator has listed user actions to use when testing if bug is fixed
- [ ] The bug is fixed
- [ ] Unit tests are added or updated

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

File: packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js [hunk 1]
```diff
@@ -41,6 +41,14 @@ const SubjectViewer = types
...
       return (!self.annotate && self.move) ? 'move' : 'annotate'
+    },
+    get disableImageToolbar () {
+      const subject = getRoot(self).subjects.active
+      const locationType = subject?.locations[self.frame].type
+      if (locationType === 'text') {
+        return true
+      }
+      return false
     }
```
[the remark this answers is not in the document]
Comment (mcbouslog) [reply packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js:52]: Sorry I'm not sure I follow. Would the view in the Subject store be:
```
get disableImageToolbar (frame) {
  if (self.locations[frame].type === 'text') {
    return true
  }
  return false
  }
}
```
Then I'd call that view in the component (i.e. MoveButtonContainer) with the subject from the subject store and the active frame from the subject viewer store?

[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js:52]: Yeah. This function only references subject locations, so it fits better in the subject model.

[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js:52]: For some reason, I thought `frame` was a property of subjects. 🤔

File: packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js [hunk 2]
```diff
@@ -41,6 +41,14 @@ const SubjectViewer = types
     get interactionMode () {
       // Default interaction mode is 'annotate'
       return (!self.annotate && self.move) ? 'move' : 'annotate'
+    },
+    get disableImageToolbar () {
+      const subject = getRoot(self).subjects.active
+      const locationType = subject?.locations[self.frame].type
+      if (locationType === 'text') {
```
[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js:48]: Probably not worth it, as that subject viewer isn’t likely to be used outside of PH-TESS.

File: packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcbouslog) [reply packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js]: No worries. I previously had the variables poorly worded as well which didn't help.

**Review by eatyourgreens — APPROVED** [review]
Changes look good to me.

**Pull request merged**